Framework/CurveFitting/src/Functions/CrystalElectricField.cpp

```diff
@@ -568,19 +564,21 @@ void diagonalise(const ComplexFortranMatrix &hamiltonian, DoubleFortranVector &e
   eigenvalues.allocate(1, dim);
   eigenvectors.allocate(1, dim, 1, dim);
   ComplexFortranMatrix h = hamiltonian;
+
   h.eigenSystemHermitian(eigenvalues, eigenvectors);
 
-  // Sort the eigenvalues in ascending order
+  // Get the indicies of the eigenvalues sorted in ascending order
   auto sortedIndices = eigenvalues.sortIndices();
+
+  // Shift the lowest energy level to 0
+  auto indexMin = static_cast<int>(sortedIndices[0] + 1); // default fortran matrix has base 1.
```

Comment:
Is it possible for `sortedIndices` to be empty?

Reply:
Doesn't look like it - `sortedIndices` is initialised using `eigenvalues.sortIndices()`. Shortly before, `eigenvalues.allocate(1, dim)` is called. The `eigenvalues` vector therefore has atleast 1 element (`dim` has to be >= 1).